10.IDR_metadata/scripts/0.preprocess_profiles.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+# Parameters
+cell_type = "SHSY5Y"
+
+
+# In[2]:
+
+
+import pathlib
+
+import numpy as np
+import pandas as pd
+import papermill as pm
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+# In[3]:
+
+
+# Parameters
+cell_type = "SHSY5Y"
+
+
+# In[4]:
+
+
+# Define inputs
+feature_file = pathlib.Path(f"../../data/{cell_type}_sc_norm.parquet")
+feature_df = pd.read_parquet(feature_file)
+
+
+# In[ ]:
+
+
+len(feature_df["Metadata_Well"].unique())
+
+
+# In[ ]:
+
+
+# replace all " " with "_" in all values of the dataframe
+feature_df = feature_df.replace(to_replace=" ", value="_", regex=True)
+
+
+# In[ ]:
+
+
+# remove uM in each row of the Metadata_inducer1_concentration column
+feature_df["Metadata_inducer1_concentration"] = feature_df[
+    "Metadata_inducer1_concentration"
+].str.replace("µM", "")
+
+
+# In[ ]:
+
+
+feature_df["Metadata_inducer1_concentration"].unique()
+
+
+# In[ ]:
+
+
+# define output file path
+feature_df_out_path = pathlib.Path(
+    f"../../data/{cell_type}_preprocess_sc_norm_no_fs.parquet"
+)
+
+
+# In[ ]:
+
+
+print(feature_df.shape)
+feature_df.head()
+
+
+# In[ ]:
+
+
+# removing costes features as they behave with great variance across all data
+feature_df = feature_df.drop(feature_df.filter(regex="Costes").columns, axis=1)
+print(feature_df.shape)
+feature_df.head()
+
+
+# In[ ]:
+
+
+# replacing '/' in treatment dosage column to avoid errors in file interpolation including such strings
+feature_df = feature_df.replace(to_replace="/", value="_per_", regex=True)
+
+
+# In[ ]:
+
+
+# replace nan values with 0
+
+columns_to_fill = [
+    "Metadata_inducer1_concentration",
+    "Metadata_inducer2_concentration",
+    "Metadata_inhibitor_concentration",
+]
+feature_df[columns_to_fill].fillna(0, inplace=True)
+
+
+# In[ ]:
+
+
+# replace all None values with 0
+feature_df["Metadata_inducer1_concentration"].fillna(0, inplace=True)
+
+
+# In[ ]:
+
+
+# create a list of columns to be converted to float
+col_list = [
+    "Metadata_inducer1_concentration",
+    "Metadata_inducer2_concentration",
+    "Metadata_inhibitor_concentration",
+]
+# loop through the list and convert each column to float
+for i in col_list:
+    feature_df[i] = feature_df[i].apply(
+        lambda x: f"{float(x):.3f}" if float(x) != 0 else float(x)
+    )
+
+
+# In[ ]:
+
+
+len(feature_df["Metadata_Well"].unique())
+
+
+# #### Combine Inducer1 and Inducer2 into one column
```

Comment:
Why is this necessary? (please answer this question in the documentation)

Reply:
Removed at this line, but updated documentation in a future line that uses this code.